test/server.cpp

```diff
@@ -310,6 +317,28 @@ TEST_F(ServerTest, BookMainPageIsRedirectedToArticleIndex)
   ASSERT_EQ("/zimfile/A/index", g->get_header_value("Location"));
 }
 
+
+TEST_F(ServerTest, RawEntry)
+{
+  auto p = zfs1_->GET("/raw/zimfile/meta/Title");
+  EXPECT_EQ(200, p->status);
+  EXPECT_EQ(p->body, std::string("Ray Charles"));
+
+  p = zfs1_->GET("/raw/zimfile/meta/Creator");
+  EXPECT_EQ(200, p->status);
+  EXPECT_EQ(p->body, std::string("Wikipedia"));
+
+  // The raw content of Ray_Charles doesn't contain a taskbar...
+  p = zfs1_->GET("/raw/zimfile/content/A/Ray_Charles");
+  EXPECT_EQ(200, p->status);
+  EXPECT_TRUE(p->body.find("taskbar") == std::string::npos);
```

Comment:
Isn't this check too weak? It may be fine for whitebox testing, but for blackbox testing you should validate that the content returned by raw endpoint hasn't undergone *any* transformations.

Reply:
I somehow agree. But we are doing whitebox testing :)
This is why I also add a test on "A/Ray_Charles" path to be sure that "taskbar" is in the returned content.
It is not easy to be sure that we have not do any transformation on a content, even more if target a blackbox testing.

Reply:
> It is not easy to be sure that we have not do any transformation on a content, even more if target a blackbox testing.

Why don't you just compare `p->body` with the same article fetched from the ZIM file directly?

Reply:
With stupid question, we have stupid answer :) Thanks. Will do.

Reply:
You were right to ask for a correct test. The code was buggy. Fixed.